oauth2/oauth2-core/src/main/java/org/springframework/security/oauth2/core/ClaimAccessor.java

```diff
@@ -159,4 +159,17 @@ default URL getClaimAsURL(String claim) {
 		}
 		return convertedValue;
 	}
+
+	/**
+	 * Returns the claim value as a {@code T} type.
+	 * It assumes that the given claim has already been converted to type {@code T}.
+	 *
+	 * @param claim the name of the claim
+	 * @param <T> the type of the claim
+	 * @return the claim value
+	 */
+	@SuppressWarnings("unchecked")
+	default <T> T getClaim(String claim) {
+		return (T) this.getClaims().get(claim);
```

Comment:
We should add `containsClaim(claim)` (same as other methods) which ensures the provided `claim` is not `null`

Reply:
Done